lib/nrf_modem_lib/nrf_modem_lib_trace.c

```diff
@@ -88,38 +191,35 @@ int nrf_modem_lib_trace_start(enum nrf_modem_lib_trace_mode trace_mode)
 
 int nrf_modem_lib_trace_process(const uint8_t *data, uint32_t len)
 {
-	if (!is_transport_initialized) {
-		return -ENXIO;
-	}
+	int ret = 0;
 
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_UART
-	/* Split RAM buffer into smaller chunks to be transferred using DMA. */
-	uint32_t remaining_bytes = len;
-	const uint32_t MAX_BUF_LEN = (1 << UARTE1_EASYDMA_MAXCNT_SIZE) - 1;
+	if (is_transport_initialized) {
+		struct trace_data_t trace_data = { .data = data, .len = len };
```

Comment:
What you are doing here is first make a copy of `trace_data` on the stack, and then copy that to the heap.
Have you considered having one copy of this structure, allocating it on the heap directly?

Reply:
Have changed it to allocate the struct on the heap directly.